purchase_orderline_images/reports/purchase_order_report.xml

```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <template
+        id="report_purchase_order_product_image"
+        inherit_id="purchase.report_purchaseorder_document"
+    >
+        <xpath
+            expr="//table[hasclass('o_main_table')]/thead/tr/th[@name='th_taxes']"
+            position="before"
+        >
+            <t t-if="o.check_show_product_image_in_purchase_report()">
```

Comment:
You can optimise performance by using a variable once. Instead of calling same function twice.
```
<t t-set="show_image" t-value="o.check_show_product_image_in_purchase_report()"
....
<t t-if="show_image">
...
```